src/node_crypto.cc

```diff
@@ -2097,6 +2099,52 @@ void SSLWrap<Base>::GetPeerCertificate(
 }
 
 
+template <class Base>
+void SSLWrap<Base>::GetFinished(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  Base* w;
+  ASSIGN_OR_RETURN_UNWRAP(&w, args.Holder());
+
+  // We cannot just pass nullptr to SSL_get_finished()
+  // because it would further be propagated to memcpy(),
+  // where the standard requirements as described in ISO/IEC 9899:2011
+  // sections 7.21.2.1, 7.21.1.2, and 7.1.4, would be violated.
+  // Thus, we use a dummy byte.
+  char dummy[1];
+  size_t len = SSL_get_finished(w->ssl_, dummy, sizeof dummy);
+  if (!len)
+    return;
+
+  char* buf = Malloc(len);
+  SSL_get_finished(w->ssl_, buf, len);
```

Comment:
Can you `CHECK_EQ(size, SSL_get_finished(...))` here and on line 2143?

Reply:
@bnoordhuis Done in the amended commit.